controllers/rack.go

```diff
@@ -319,6 +319,12 @@ func (r *SingleClusterReconciler) reconcileRack(
 		return reconcileError(err)
 	}
 
+	// For each block volume removed from a namespace, pod status dirtyVolumes is appended with that volume name.
+	// For each file removed from a namespace, it is deleted right away.
+	if err := r.handleNSOrDeviceRemoval(rackState); err != nil {
```

Comment:
See if files can be removed after the pod restart. If someone restores the older namespace then the namespace file should be there to reuse.

Reply:
we can move whole handleNSOrDeviceRemoval function after restart, which results in patching dirtyVolumes and files delete operation will be done after pod restart. 
Doing this, will limit user to perform disk usage operation in single CR update.